### PR TITLE
Update and Fix ROSE Package

### LINKED ISSUE #12983: Installation issue: Rose
Trying to install chill, its dependency rose 9.10.0 fails to build with the error output below. I think the critical error is failure to download EDG but there are some other potential issues mentioned. I am building with gcc-7.3.0 on a RHEL6 system. 

```
==> Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j12'

18 errors found in build log:
     444     Successfully ran a simple program with the backend C compiler: /storage/users/wspear/bin/SPACK/spack/lib/spack/env/gcc
             /gcc
     445     => cross-compiling... false
     446     build input file for backend compiler
     447     int main(int argc, char** argv){ asm("nop");}
     448     run backend compiler on input file: /storage/users/wspear/bin/SPACK/spack/lib/spack/env/gcc/gcc -std=c99 -Werror=impli
             cit-function-declaration -c conftest_asm.c
     449     conftest_asm.c: In function 'main':
  >> 450     conftest_asm.c:1:34: error: implicit declaration of function 'asm' [-Werror=implicit-function-declaration]
     451      int main(int argc, char** argv){ asm("nop");}
     452                                       ^~~
     453     cc1: some warnings being treated as errors
     454     status = 1
     455     true case: set asm_ok=no
     456     checking does the backend C compiler (/storage/users/wspear/bin/SPACK/spack/lib/spack/env/gcc/gcc) support asm stateme
             nts... no

     ...

     460     status = 0
     461     false case: set underscore_asm_ok=yes
     462     checking does the backend C compiler (/storage/users/wspear/bin/SPACK/spack/lib/spack/env/gcc/gcc) support __asm__ sta
             tements... yes
     463     build input file for backend compiler
     464     asm(".symver ff_av_gettime,av_gettime@LIBAVFORMAT_54");
     465     run backend compiler on input file: /storage/users/wspear/bin/SPACK/spack/lib/spack/env/gcc/gcc -std=c99 -Werror=impli
             cit-function-declaration -c conftest_long_string_asm.c
  >> 466     conftest_long_string_asm.c:1:5: error: expected declaration specifiers or '...' before string constant
     467      asm(".symver ff_av_gettime,av_gettime@LIBAVFORMAT_54");
     468          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     469     status = 1
     470     true case: set long_string_asm_ok=no
     471     checking does the backend C compiler (/storage/users/wspear/bin/SPACK/spack/lib/spack/env/gcc/gcc) support asm stateme
             nts... no
     472     checking for C++ compiler vendor... gnu

     ...

     1361    checking for POET... In ROSE SUPPORT MACRO: with_poet no
     1362    Skipping installation for POET!
     1363    Setup installation for POET in ROSE! with_poet = no
     1364    checking for fgrep... /bin/grep -F
     1365    checking for objdump... objdump
     1366    checking for a sed that does not truncate output... (cached) sed
  >> 1367    checking dynamic linker characteristics... /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spac
             k-src/configure: line 23835:  /storage/users/wspear/bin/SPACK/spack/opt/spack/linux-rhel6-westmere/gcc-6.1.0/gcc-7.3.0
             -pjk43on4l6bychpz2rowuk6somtudjdm/lib/gcc/x86_64-pc-linux-gnu/7.3.0/:/storage/users/wspear/bin/SPACK/spack/opt/spack/l
             inux-rhel6-westmere/gcc-6.1.0/gcc-7.3.0-pjk43on4l6bychpz2rowuk6somtudjdm/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../..
             /x86_64-pc-linux-gnu/lib/x86_64-pc-linux-gnu/7.3.0/:/storage/users/wspear/bin/SPACK/spack/opt/spack/linux-rhel6-westme
             re/gcc-6.1.0/gcc-7.3.0-pjk43on4l6bychpz2rowuk6somtudjdm/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../../x86_64-pc-linux-
             gnu/lib/../lib64/:/storage/users/wspear/bin/SPACK/spack/opt/spack/linux-rhel6-westmere/gcc-6.1.0/gcc-7.3.0-pjk43on4l6b
             ychpz2rowuk6somtudjdm/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../x86_64-pc-linux-gnu/7.3.0/:/storage/users/wspear/bin/
             SPACK/spack/opt/spack/linux-rhel6-westmere/gcc-6.1.0/gcc-7.3.0-pjk43on4l6bychpz2rowuk6somtudjdm/lib/gcc/x86_64-pc-linu
             x-gnu/7.3.0/../../../../lib64/:/lib/x86_64-pc-linux-gnu/7.3.0/:/lib/../lib64/:/usr/lib/x86_64-pc-linux-gnu/7.3.0/:/usr
             /lib/../lib64/:/storage/users/wspear/bin/SPACK/spack/opt/spack/linux-rhel6-westmere/gcc-6.1.0/gcc-7.3.0-pjk43on4l6bych
             pz2rowuk6somtudjdm/lib/gcc/x86_64-pc-linux-gnu/7.3.0/../../../../x86_64-pc-linux-gnu/lib/:/storage/users/wspear/bin/SP
             ACK/spack/opt/spack/linux-rhel6-westmere/gcc-6.1.0/gcc-7.3.0-pjk43on4l6bychpz2rowuk6somtudjdm/lib/gcc/x86_64-pc-linux-
             gnu/7.3.0/../../../:/lib/:/usr/lib/: No such file or directory
     1368    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 23876: : command not 
             found
     1369    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 23883: : command not 
             found
     1370    GNU/Linux ld.so
     1371    checking how to run the C++ preprocessor... /storage/users/wspear/bin/SPACK/spack/lib/spack/env/gcc/g++ -E
     1372    checking for flex... flex
     1373    checking lex output file root... lex.yy

     ...

     1544    configure:     PYTHON_VERSION_PATCH_VERSION =
     1545    configure:     ROSE_USE_PYTHON              = /usr/local/packages/python/2.7.7/bin/python2.7
     1546    
     1547    LLVM
     1548    In ROSE SUPPORT MACRO: with_llvm no
     1549    Skipping use of LLVM support (and opt)!
  >> 1550    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33264: /bin/llvm-conf
             ig: No such file or directory
  >> 1551    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33265: /bin/llvm-conf
             ig: No such file or directory
  >> 1552    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33284: /bin/llvm-conf
             ig: No such file or directory
  >> 1553    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33285: /bin/llvm-conf
             ig: No such file or directory
  >> 1554    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33287: /bin/llvm-conf
             ig: No such file or directory
  >> 1555    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33288: /bin/llvm-conf
             ig: No such file or directory
  >> 1556    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33289: /bin/llvm-conf
             ig: No such file or directory
  >> 1557    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/configure: line 33290: /bin/llvm-conf
             ig: No such file or directory
     1558    checking for Wine -- Wine is not an emulator... In ROSE SUPPORT MACRO: with_wine no
     1559    Skipping use of Wine for Windows Binary Analysis!
     1560    checking for specification of EDG union/struct debugging support... In ROSE SUPPORT MACRO: enable_edg_union_struct_deb
             ugging no
     1561    Skipping use of EDG Union/Struct debugging support!
     1562    checking for OpenMP using Omni runtime library... In ROSE SUPPORT MACRO: with_omni_omp_runtime_support no
     1563    Skipping use of Omni OpenMP Runtime support!

     ...

     1797     int main() {return 0;}
     1798         ^~~~
     1799    /tmp/rose_support_i386_28343.c:1:5: warning: function declaration isn't a prototype [-Wstrict-prototypes]
     1800     int main() {return 0;}
     1801         ^~~~
     1802    /storage/users/wspear/bin/SPACK/spack/opt/spack/linux-rhel6-westmere/gcc-6.1.0/gcc-7.3.0-pjk43on4l6bychpz2rowuk6somtud
             jdm/lib/gcc/x86_64-pc-linux-gnu/7.3.0/crtbegin.o: could not read symbols: File in wrong format
  >> 1803    collect2: error: ld returned 1 exit status
     1804    no
     1805    In ROSE SUPPORT MACRO: with_sat no
     1806    Skipping use of internal SAT Solver support!
     1807    with_sat = no
     1808    checking for IntelPin... In ROSE SUPPORT MACRO: with_IntelPin no
     1809    Skipping use of Intel Pin package support!

     ...

     4356    make[5]: Entering directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-bui
             ld/src/ROSETTA/src'
     4357    make[5]: Nothing to be done for `all-am'.
     4358    make[5]: Leaving directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-buil
             d/src/ROSETTA/src'
     4359    make[4]: Leaving directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-buil
             d/src/ROSETTA/src'
     4360    /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/scripts/edg-generate-sig /tmp/wspear/
             spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src ../../.. >edg-generate-sig.output.tmp
     4361    diff -q edg-generate-sig.output edg-generate-sig.output.tmp || mv edg-generate-sig.output.tmp edg-generate-sig.output
  >> 4362    diff: edg-generate-sig.output: No such file or directory
     4363    make[4]: Entering directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-bui
             ld/src/frontend/CxxFrontend'
     4364    make[4]: Leaving directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-buil
             d/src/frontend/CxxFrontend'
     4365    Downloading EDG binary tarball: roseBinaryEDG-4-12-x86_64-pc-linux-gnu-gnu-7.3-4.5d9f1499e6eb31c0440db039b45336adf4e5a
             c6b.tar.gz
  >> 4366    error: the EDG binary tarball is not available (or the web site is not reachable).
     4367    Please contact rose-public@nersc.gov to troubleshoot this error.
  >> 4368    make[3]: *** [EDG.tar.gz] Error 1
     4369    make[3]: Leaving directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-buil
             d/src/frontend/CxxFrontend'
  >> 4370    make[2]: *** [all-recursive] Error 1
     4371    make[2]: Leaving directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-buil
             d/src/frontend/CxxFrontend'
  >> 4372    make[1]: *** [all-recursive] Error 1
     4373    make[1]: Leaving directory `/tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-src/rose-buil
             d/src/frontend'
  >> 4374    make: *** [all-recursive] Error 1

See build log for details:
  /tmp/wspear/spack-stage/rose-0.9.10.0-36tk7qmn7j7hv5snpcsplwglv65nkx4z/spack-build-out.txt
```

### PR DESCRIPTION
**Pull request by pinnown**
- Added newest version of ROSE
- Modified to use new method of getting EDG
- Updated dependencies
- Removed older versions as they depend on resources that are no longer available or will be removed soon.

Closes #12983

### CODE REVIEW

File: var/spack/repos/builtin/packages/rose/package.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 # -----------------------------------------------------------------------------
-# Author: Justin Too
+# Author: Nathan Pinnow
 # -----------------------------------------------------------------------------
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rose/package.py]: This comment isn't actually used by anything, would you like to remove it and add yourself under:
```python
maintainers = ['pinnown']
```
within the class definition? See bullet point 3 under https://spack.readthedocs.io/en/latest/packaging_guide.html#creating-editing-packages for a description of what this means.

File: var/spack/repos/builtin/packages/rose/package.py [hunk 2]
```diff
@@ -68,12 +95,10 @@ class Rose(AutotoolsPackage):
...
 
     # Use spack install cxxflags=-std=c++11
-    variant("cxx11", default=False, description="Enable c++11 language support")
+    variant("cxx11", default=True, description="Enable c++11 language support")
 
     variant("fortran", default=False, description="Enable fortran language support")
-
-    variant("java", default=False, description="Enable java language support")
-    depends_on("jdk", when="+java")
+    depends_on("openjdk@1.8.0_202-b08", when="+fortran")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/rose/package.py]: Does it work with any other Java provider? What version of Java is required? You should be able to do something like:
```python
depends_on('java@8', when='+fortran')
```
and have it use any Java 8 provider

Comment (pinnown) [reply var/spack/repos/builtin/packages/rose/package.py]: We have only test with openjdk 1.8 but there is no reason it shouldn't work with other java 8.

File: var/spack/repos/builtin/packages/rose/package.py (hunk 1 above)
Comment (pinnown) [reply var/spack/repos/builtin/packages/rose/package.py]: Switched authors to maintainers.

File: var/spack/repos/builtin/packages/rose/package.py (hunk 2 above)
Comment (pinnown) [reply var/spack/repos/builtin/packages/rose/package.py]: Changed dependency to be on java@8.

**Comment by adamjstewart** [comment]
@pinnown thanks for volunteering to be a maintainer of the `rose` package! Could you take a look at #12983 and see if this PR fixes it?

**Pull request merged**